### PR TITLE
tests with higher level chaining api

### PR DESCRIPTION
**Pull request by madhur-ob**
This should only be merged after https://github.com/Netflix/metaflow/pull/1732

Notes: we need `typeguard` which is apparently not available for Python 3.5 and 3.6

https://pypi.org/project/typeguard/

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -43,6 +43,9 @@ jobs:
     - name: Execute Python tests
       run: tox
 
+    - name: Execute Python tests with low level chaining API
+      run: tox -e chaining
```
Comment (savingoyal) [inline .github/workflows/test.yml:47]: it might be better not to introduce another test in the yaml but to embed the test execution in the context

Comment (savingoyal) [reply .github/workflows/test.yml:47]: you could introduce an `executor` keyword in the context that takes in two options - `cli` and `api` - similar to `checker`

Comment (savingoyal) [reply .github/workflows/test.yml:47]: the tests will go against the high level API and not the low level API but hopefully that's a simple enough change

Comment (madhur-ob) [reply .github/workflows/test.yml]: done, added lower level api as an executor for now, can switch to higher level later once the changes for UX are in.

**Comment by savingoyal** [comment]
#1859 merged with a fix

**Pull request closed without merging**